app.rb

```diff
@@ -0,0 +1,83 @@
+require './person'
+require './book'
+require './classroom'
+require './student'
+require './teacher'
+require './rental'
+
+class App
+  attr_accessor :people, :books, :rentals
+
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_books
+    @books.each { |book| puts "Title: \"#{book.title}\", Author: #{book.author}" }
+  end
+
+  def list_people
+    @people.each { |person| puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
+  end
+
+  def create_person
+    print 'Do you want to Create a student (1) or a teacher (2)? [Input the number]: '
+    select_person = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp.to_i
+    case select_person
+    when 1
+      print 'Has parent permission? [Y/N]: '
+      permission = gets[0].capitalize
+      permission = permission == 'Y'
+      @people << Student.new(nil, age, name, permission)
+    when 2
+      print 'Specialization: '
+      specialization = gets.chomp
+      @people << Teacher.new(age, name, specialization)
+    end
+    puts 'Person created successfully'
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    @books << Book.new(title, author)
+    puts 'Book created successfully'
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number'
+    @books.each_with_index { |book, index| puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}" }
+    book_number = gets.chomp.to_i
+    puts 'Select a person from the following list by number (not id)'
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    person_number = gets.chomp.to_i
+    print 'Date: '
+    date = gets.chomp
+    @rentals << Rental.new(date, @people[person_number], @books[book_number])
+    puts 'Rental created successfully'
+  end
+
```

Comment:
- [ ] Everything is working well in your app except for the rentals. When trying to add the date for a rental the user gets an error as you can see in the caption below:

<img width="656" alt="Screenshot 2023-09-08 at 5 39 14 PM" src="https://github.com/tajulafreen/School_library/assets/53347966/14ec7615-4d70-4a01-a973-b60de8b5d602">

Kindly check your parameters for that option and fix the issue (hint: you have to choose a format for the date).